servicetalk-dns-discovery-netty/src/test/java/io/servicetalk/dns/discovery/netty/DefaultDnsClientTest.java

```diff
@@ -593,26 +594,36 @@ void unknownHostDiscover() throws Exception {
         assertThat(subscriber.awaitOnError(), instanceOf(UnknownHostException.class));
     }
 
-    @Test
-    void unknownHostAfterKnownDiscover() throws Exception {
-        setup();
-        final String targetDomain1 = "sd.domain.com";
-        final String ip1 = nextIp();
+    @ParameterizedTest(name = "{displayName} [{index}] cache={0}")
+    @ValueSource(booleans = {false, true})
+    void unknownHostAfterKnownDiscover(boolean nxInvalidation) throws Exception {
+        boolean restore = NX_DOMAIN_INVALIDATES;
+        try {
+            NX_DOMAIN_INVALIDATES = nxInvalidation;
```

Comment:
Ideally, we should have all constants as `final`. Alternatively, consider adding a pkg-private builder method that will modify the value and then pass it to `DefaultDnsClient` as a ctor argument instead of direct dependency on a non-final constant from the builder